### ISSUE TITLE
linux/arm64 support

### ISSUE THREAD

**Issue by Hanse00** [super-linter/super-linter#4498]
Given that many of us work in an arm64 context these days, it would be great if the linter had native linux\arm64 platform builds as well as the existing linux\amd64 builds. This idea was previously brought up in #1956, I'm marking it here to formally commit us to exploring the idea.

@BretFisher and @denisa have already done a fair bit of legwork in exploring the technical barriers to providing this solution, namely:

* Some of our upstream container images currently do not offer arm64 builds, therefore we would either need to contribute those upstream, or move to either building certain things from source, or at least depending on their binary versions, rather than the containers.
* A number of scripts used by the linter have hardcoded assumptions regarding architecture, those would need to be abstracted. For what it's worth many of those scripts could use a second pass either way, they're not great.

**Comment by Hanse00** [comment]
After a bit of discussion with @zkoppert, our initial pitch for an approach is as follows:

* Move away from depending on third party container images, and instead build our own (For both x86 and ARM) based on either binary distributions or source depending on the individual dependency.
* Continue to (and perhaps even to a larger degree) build the finished container based on a set of container inputs, so that we only need to rebuild the parts that change, and can cache the rest

**Comment by BretFisher** [comment]
I'm happy to (at a minimum) review anything, including WIP PRs. I do have a few thoughts:

1. With so many tools in so many places, you likely won't end up with a single method for injecting all tool bins and libs. The list of tools without arm64 images [is getting smaller](https://github.com/super-linter/super-linter/discussions/1956#discussioncomment-5625180). Are you suggesting the Dockerfile replace the `COPY --from` for a different method (bin download or build)? That seems like it would add loc and complexity, but I'm interested if you think it'll be more reliable or simpler some other way.

2. On your build caching, Docker can do that itself without needing to build a different solution. I see we're using BuildKit (Docker's [Official GHA](https://github.com/marketplace/actions/build-and-push-docker-images)). Good. First, we should enable the GitHub caching, see [my example here](https://github.com/BretFisher/docker-build-workflow/blob/546af227e6dc662d5c84027cd3233e78b6f14691/.github/workflows/reusable-docker-build.yaml#L181-L182) and [docs here](https://docs.docker.com/build/ci/github-actions/cache/#github-cache). It's just a 2-line change in each build step. Next, the `COPY --from` commands should be runnable [in parallel for way faster builds](https://docs.docker.com/engine/reference/builder/#copy---link) with `COPY --link` and even injected in rebuilds without needing (or breaking) build cache. It's dope but rather new so many don't know about it.

I'm happy to join a real-time discussion elsewhere if you want to plan something.

**Comment by denisa** [comment]
I believe it is worthwhile to first work with the upstream project. In my experience, most projects are open to publish/host official multi-architecture images, but don’t see it as a priority, so they would accept contributions.

This better for the OSS community, and take out the long term maintenance effort involved if we were to have our own images.
